test-suite/Math/NumberTheory/DiophantineTests.hs

```diff
@@ -0,0 +1,22 @@
+-- Tests for Math.NumberTheory.Diophantine
+
+{-# LANGUAGE CPP       #-}
+
+{-# OPTIONS_GHC -fno-warn-type-defaults #-}
+
+module Math.NumberTheory.DiophantineTests
+  ( testSuite
+  ) where
+
+import Test.Tasty
+
+import Math.NumberTheory.Diophantine
+import Math.NumberTheory.TestUtils
+
+cornacchiaTest :: Positive Integer -> Positive Integer -> Bool
+cornacchiaTest (Positive d) (Positive a) = all checkSoln (cornacchia d m)
```

Comment:
Such test will pass even if we define `cornacchia _ _ = []`. Please write a reference implementation, searching for solutions using brute force, and compare the output of `cornacchia` against this reference.